ClassNotFoundException in ValueTypes functional extended test on Windows

Failure link
------------

https://openj9-jenkins.osuosl.org/job/Test_openjdkValhalla_j9_extended.functional_x86-64_windows_valhalla_Personal/1/tapResults/



Failure output (captured from console output)
---------------------------------------------
```
cmdLineTester_unflattenedvaluetypeddrtests_0 Start Time: Fri Oct  1 07:47:56 2021 Epoch Time (ms): 1633092476447
variation: NoOptions
JVM_OPTIONS:  
Start SysV clean up for user:all 
ipcs: msgctl: Function not implemented
Error: could not execute ipcs! at F:/Users/jenkins/workspace/Test_openjdkValhalla_j9_extended.functional_x86-64_windows_valhalla_Personal_testList_1/aqa-tests///..//jvmtest\TestConfig\scripts\tools\sysvcleanup.pl line 87.
JVMJ9VM193W Since Java 13 -Xverify:none and -noverify were deprecated for removal and may not be accepted options in the future.
*** Starting test suite: J9 Value Type ddr test with flattening disabled ***
Executing command: rm -f j9core.dmp

Executing command: rm -f core*

Testing: Create core file
Test start time: 2021/10/01 07:47:57 Central Standard Time
Running command: "F:/Users/jenkins/workspace/Test_openjdkValhalla_j9_extended.functional_x86-64_windows_valhalla_Personal_testList_1/openjdkbinary/j2sdk-image\\bin\\java"   -Xmx24m -Xint -Xverify:none --add-opens java.base/jdk.internal.misc=ALL-UNNAMED -cp F:/Users/jenkins/workspace/Test_openjdkValhalla_j9_extended.functional_x86-64_windows_valhalla_Personal_testList_1/aqa-tests///..//jvmtest\TKG\lib\asm-all.jar:F:/Users/jenkins/workspace/Test_openjdkValhalla_j9_extended.functional_x86-64_windows_valhalla_Personal_testList_1/aqa-tests///..//jvmtest\TKG\lib\jcommander.jar:F:/Users/jenkins/workspace/Test_openjdkValhalla_j9_extended.functional_x86-64_windows_valhalla_Personal_testList_1/aqa-tests///..//jvmtest\TKG\lib\testng.jar:F:/Users/jenkins/workspace/Test_openjdkValhalla_j9_extended.functional_x86-64_windows_valhalla_Personal_testList_1/aqa-tests///..//jvmtest\functional\Valhalla\ValhallaTests.jar org.openj9.test.lworld.DDRValueTypeTest
Time spent starting: 16 milliseconds
Time spent executing: 235 milliseconds
Test result: FAILED
 [ERR] JVMJ9VM193W Since Java 13 -Xverify:none and -noverify were deprecated for removal and may not be accepted options in the future.
 [ERR] Error: Could not find or load main class org.openj9.test.lworld.DDRValueTypeTest
 [ERR] Caused by: java.lang.ClassNotFoundException: org.openj9.test.lworld.DDRValueTypeTest
>> Success condition was not found: [Output match: System dump written]
>> Required condition was not found: [Output match: System dump written to ]
>> Failure condition was not found: [Output match: Exception caught!]
```

This is a ValueTypes specific test. There is a classpath issue here.

--- a/test/functional/cmdLineTests/valuetypeddrtests/flattened32bitRefBackfillTest.xml
+++ b/test/functional/cmdLineTests/valuetypeddrtests/flattened32bitRefBackfillTest.xml
@@ -26,7 +26,7 @@
 
 <suite id="J9 Value Type ddr test with flattening enabled" timeout="600">
 	<variable name="ARGS" value="-XX:ValueTypeFlatteningThreshold=999999 -XX:+EnableArrayFlattening -Xverify:none --add-opens java.base/jdk.internal.misc=ALL-UNNAMED" />
-	<variable name="JARS" value="-cp $ASMJAR$:$JCOMMANDERJAR$:$TESTNGJAR$:$VALUETYPEJAR$" />
+	<variable name="JARS" value="-cp $ASMJAR$$CPDL$$JCOMMANDERJAR$$CPDL$$TESTNGJAR$$CPDL$$VALUETYPEJAR$" />
 	<variable name="PROGRAM" value="org.openj9.test.lworld.DDRBackfillLayouTest" />
 	<variable name="DUMPFILE" value="j9core.dmp" />
 

--- a/test/functional/cmdLineTests/valuetypeddrtests/flattenedvaluetypeddrtests.xml
+++ b/test/functional/cmdLineTests/valuetypeddrtests/flattenedvaluetypeddrtests.xml
@@ -27,7 +27,7 @@
 <suite id="J9 Value Type ddr test with flattening enabled" timeout="600">
 	<variable name="ARGS" value="-Xint -XX:ValueTypeFlatteningThreshold=999999 -XX:+EnableArrayFlattening -Xverify:none --add-opens java.base/jdk.internal.misc=ALL-UNNAMED" />
 	<variable name="ARGS_NOCR" value="-Xint -Xnocompressedrefs -XX:ValueTypeFlatteningThreshold=999999 -XX:+EnableArrayFlattening -Xverify:none --add-opens java.base/jdk.internal.misc=ALL-UNNAMED" />
-	<variable name="JARS" value="-cp $ASMJAR$:$JCOMMANDERJAR$:$TESTNGJAR$:$VALUETYPEJAR$" />
+	<variable name="JARS" value="-cp $ASMJAR$$CPDL$$JCOMMANDERJAR$$CPDL$$TESTNGJAR$$CPDL$$VALUETYPEJAR$" />
 	<variable name="PROGRAM" value="org.openj9.test.lworld.DDRValueTypeTest" />
 	<variable name="DUMPFILE" value="j9core.dmp" />
 

--- a/test/functional/cmdLineTests/valuetypeddrtests/playlist.xml
+++ b/test/functional/cmdLineTests/valuetypeddrtests/playlist.xml
@@ -37,6 +37,7 @@ SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-excepti
 			-DJCOMMANDERJAR=$(Q)$(JVM_TEST_ROOT)$(D)TKG$(D)lib$(D)jcommander.jar$(Q) \
 			-DTESTNGJAR=$(Q)$(JVM_TEST_ROOT)$(D)TKG$(D)lib$(D)testng.jar$(Q) \
 			-DVALUETYPEJAR=$(Q)$(JVM_TEST_ROOT)$(D)functional$(D)Valhalla$(D)ValhallaTests.jar$(Q) \
+			-DCPDL=$(Q)$(P)$(Q) \
 			-jar $(CMDLINETESTER_JAR) \
 			-config $(Q)$(TEST_RESROOT)$(D)flattenedvaluetypeddrtests.xml$(Q) \
 			-outputLimit 1000 -explainExcludes -nonZeroExitWhenError; \
@@ -74,6 +75,7 @@ SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-excepti
 			-DJCOMMANDERJAR=$(Q)$(JVM_TEST_ROOT)$(D)TKG$(D)lib$(D)jcommander.jar$(Q) \
 			-DTESTNGJAR=$(Q)$(JVM_TEST_ROOT)$(D)TKG$(D)lib$(D)testng.jar$(Q) \
 			-DVALUETYPEJAR=$(Q)$(JVM_TEST_ROOT)$(D)functional$(D)Valhalla$(D)ValhallaTests.jar$(Q) \
+			-DCPDL=$(Q)$(P)$(Q) \
 			-jar $(CMDLINETESTER_JAR) \
 			-config $(Q)$(TEST_RESROOT)$(D)unflattenedvaluetypeddrtests.xml$(Q) \
 			-outputLimit 1000 -explainExcludes -nonZeroExitWhenError; \
@@ -112,6 +114,7 @@ SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-excepti
 			-DJCOMMANDERJAR=$(Q)$(JVM_TEST_ROOT)$(D)TKG$(D)lib$(D)jcommander.jar$(Q) \
 			-DTESTNGJAR=$(Q)$(JVM_TEST_ROOT)$(D)TKG$(D)lib$(D)testng.jar$(Q) \
 			-DVALUETYPEJAR=$(Q)$(JVM_TEST_ROOT)$(D)functional$(D)Valhalla$(D)ValhallaTests.jar$(Q) \
+			-DCPDL=$(Q)$(P)$(Q) \
 			-jar $(CMDLINETESTER_JAR) \
 			-config $(Q)$(TEST_RESROOT)$(D)flattened32bitRefBackfillTest.xml$(Q) \
 			-outputLimit 1000 -explainExcludes -nonZeroExitWhenError; \
@@ -150,6 +153,7 @@ SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-excepti
 			-DJCOMMANDERJAR=$(Q)$(JVM_TEST_ROOT)$(D)TKG$(D)lib$(D)jcommander.jar$(Q) \
 			-DTESTNGJAR=$(Q)$(JVM_TEST_ROOT)$(D)TKG$(D)lib$(D)testng.jar$(Q) \
 			-DVALUETYPEJAR=$(Q)$(JVM_TEST_ROOT)$(D)functional$(D)Valhalla$(D)ValhallaTests.jar$(Q) \
+			-DCPDL=$(Q)$(P)$(Q) \
 			-jar $(CMDLINETESTER_JAR) \
 			-config $(Q)$(TEST_RESROOT)$(D)flattened32bitRefBackfillTest.xml$(Q) \
 			-outputLimit 1000 -explainExcludes -nonZeroExitWhenError; \

--- a/test/functional/cmdLineTests/valuetypeddrtests/unflattenedvaluetypeddrtests.xml
+++ b/test/functional/cmdLineTests/valuetypeddrtests/unflattenedvaluetypeddrtests.xml
@@ -26,7 +26,7 @@
 
 <suite id="J9 Value Type ddr test with flattening disabled" timeout="600">
 	<variable name="ARGS" value="-Xint -Xverify:none --add-opens java.base/jdk.internal.misc=ALL-UNNAMED" />
-	<variable name="JARS" value="-cp $ASMJAR$:$JCOMMANDERJAR$:$TESTNGJAR$:$VALUETYPEJAR$" />
+	<variable name="JARS" value="-cp $ASMJAR$$CPDL$$JCOMMANDERJAR$$CPDL$$TESTNGJAR$$CPDL$$VALUETYPEJAR$" />
 	<variable name="PROGRAM" value="org.openj9.test.lworld.DDRValueTypeTest" />
 	<variable name="DUMPFILE" value="j9core.dmp" />
 

